src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            List<? extends Machine> machines = new BuldozerProducer().get();
+            return new ArrayList<>(machines);
+        } else if (type == Excavator.class) {
+            List<? extends Machine> machines = new ExcavatorProducer().get();
+            return new ArrayList<>(machines);
+        } else if (type == Truck.class) {
+            List<? extends Machine> machines = new TruckProducer().get();
+            return new ArrayList<>(machines);
```

Comment:
no need to create new `Producer` each time you want to call `.get()` method on it, make this producer as a class `private final` field 
```suggestion
           return new ArrayList<>(truckProducer.get());
```